src/components/accession2/edit/ViabilityModal.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect } from 'react';
+import strings from 'src/strings';
+import Button from 'src/components/common/button/Button';
+import DialogBox from 'src/components/common/DialogBox/DialogBox';
+import { Box, Grid, Link, Typography } from '@mui/material';
+import { Textfield } from '@terraware/web-components';
+import { Accession2, updateAccession2 } from 'src/api/accessions2/accession';
+import useForm from 'src/utils/useForm';
+import useSnackbar from 'src/utils/useSnackbar';
+import { preventDefaultEvent } from '@terraware/web-components/utils';
+
+export interface ViabilityDialogProps {
+  open: boolean;
+  accession: Accession2;
+  onClose: () => void;
+  reload: () => void;
+  setNewViabilityTestOpened: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export default function ViabilityDialog(props: ViabilityDialogProps): JSX.Element {
+  const { onClose, open, accession, reload, setNewViabilityTestOpened } = props;
+
+  const [record, setRecord, onChange] = useForm(accession);
+  const snackbar = useSnackbar();
+
+  const saveQuantity = async () => {
+    const response = await updateAccession2(record);
+    if (response.requestSucceeded) {
+      reload();
+      onCloseHandler();
+    } else {
+      snackbar.toastError();
+      setRecord(accession);
```

Comment:
if we skip this and the closehandler, user can continue trying.
Also, we need to add error handling like in the mocks (if percentage is > 100 etc.)